packages/form-state-manager/src/utils/manager-api.ts

```diff
@@ -220,6 +260,15 @@ const createManagerApi: CreateManagerApi = ({ onSubmit, clearOnUnmount, initiali
         }
       }
     };
+
+    if (state.fieldListeners[field.name]?.count === 1) {
```

Comment:
What does this condition? :thinking:

Reply:
We don't want to create a new watcher instance if there are multiple fields with the same name. So its created only once.